rust_src/src/keymap.rs

```diff
@@ -16,6 +16,45 @@ pub fn Ctl(c: char) -> i32 {
     (c as i32) & 0x1f
 }
 
+/// Construct and return a new keymap, of the form (keymap CHARTABLE . ALIST).
+/// CHARTABLE is a char-table that holds the bindings for all characters
+/// without modifiers.  All entries in it are initially nil, meaning
+/// "command undefined".  ALIST is an assoc-list which holds bindings for
+/// function keys, mouse events, and any other things that appear in the
+/// input stream.  Initially, ALIST is nil.
+///
+/// The optional arg STRING supplies a menu name for the keymap
+/// in case you use it as a menu with `x-popup-menu'.
+#[lisp_fn(min = "0")]
+pub fn make_keymap(string: LispObject) -> LispObject {
+    let tail: LispObject;
+    if !string.is_nil() {
+        tail = LispObject::from_raw(unsafe { list1(string.to_raw()) });
+    } else {
+        tail = LispObject::constant_nil();
+    }
+
+    let char_table = unsafe { Fmake_char_table(Qkeymap, LispObject::constant_nil().to_raw()) };
+    LispObject::from_raw(unsafe { Fcons(Qkeymap, Fcons(char_table, tail.to_raw())) })
+}
+
+/// Return t if OBJECT is a keymap.
+///
+/// A keymap is a list (keymap . ALIST),
+/// or a symbol whose function definition is itself a keymap.
+/// ALIST elements look like (CHAR . DEFN) or (SYMBOL . DEFN);
+/// a vector of densely packed bindings for small character codes
+/// is also allowed as an element.
+#[lisp_fn]
+pub fn keymapp(object: LispObject) -> LispObject {
+    let map = unsafe { LispObject::from_raw(get_keymap(object.to_raw(), false, false)) };
+    if map.is_nil() {
```

Comment:
Just return `bool` here:

    let map = .....
    map.is_not_nil()

Reply:
Oh, nice. That is indeed better.